src/cwt/claim/ietf_token_status.rs

```diff
@@ -0,0 +1,151 @@
+//! Claims from the IETF Token Status List spec.
+
+use super::{Claim, Error as ClaimError, Key};
+use serde::{Deserialize, Serialize};
+use serde_cbor::Value;
+use std::collections::BTreeMap;
+
+/// Status Claim for the ReferencedToken.
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, PartialOrd, Eq, Ord, Hash)]
+pub struct ReferencedTokenStatus {
+    pub status_list: StatusListReference,
+}
+
+/// Claim describing the location of the Referenced Token in the Status List.
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, PartialOrd, Eq, Ord, Hash)]
+pub struct StatusListReference {
+    pub idx: StatusListIndex,
+    pub uri: StatusListUri,
+}
+
+/// Index to check for status information in the Status List for the Referenced Token.
+#[derive(Copy, Clone, Debug, Deserialize, Serialize, PartialEq, PartialOrd, Eq, Ord, Hash)]
+pub struct StatusListIndex(pub u32);
+
+/// The URI where one can fetch the Status List or Status List Token
+/// containing the status information about the ReferencedToken.
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, PartialOrd, Eq, Ord, Hash)]
+pub struct StatusListUri(pub String);
```

Comment:
I know a great URI library called [`iref`](https://github.com/timothee-haudebourg/iref) that provides a nice [`UriBuf`](https://docs.rs/iref/latest/iref/uri/struct.UriBuf.html) type you can use instead of `StatusListUri` (or inside it if you want to keep a newtype). It has `serde` support with the `serde` feature. It's already used everywhere in ssi. I may be biased here :smile:

Reply:
yeah wanted to get your thoughts on this -- for now I left it as open as possible to not introduce an additional dependency & allow consumers of this library to choose whatever underlying url/uri representation they want, but maybe going with iref makes sense here 🤔

Reply:
I would hold off on this until we decide to merge this into `ssi`.